### PR TITLE
Blacklight searches and displays content from Solr, ref #17 #14 #13 #12

### PR DESCRIPTION
**Pull request by awead**
This addresses a number of issues, potentially solving them. Here's a short screencast that shows what's working:
https://dl.dropboxusercontent.com/u/9554743/catalog-demo.m4v

### CODE REVIEW

File: app/controllers/catalog_controller.rb [hunk 1]
```diff
@@ -6,115 +6,115 @@ class CatalogController < ApplicationController
...
-    # as well as in URLs -- so changing it after deployment may break bookmarked
-    # urls.  A display label will be automatically calculated from the :key,
-    # or can be specified manually to be different.
+    # Search configurations
 
+    # This is the first option in the drop-down menu, and therefore, the default.
     config.add_search_field('all_fields', label: 'All Fields', include_in_advanced_search: false) do |field|
-      all_names = config.show_fields.values.map(&:field).join(" ")
+      all_names = config.index.search_fields.join(" ")
       title_name = 'title_ssi'
```
Comment (cam156) [inline app/controllers/catalog_controller.rb]: If this line needed with the new change?

File: spec/support/matchers/blacklight.rb [hunk 2]
```diff
@@ -0,0 +1,25 @@
+RSpec::Matchers.define :have_blacklight_field do |field|
```
Comment (cam156) [inline spec/support/matchers/blacklight.rb]: :sunglasses:

File: app/controllers/catalog_controller.rb (hunk 1 above)
Comment (awead) [inline app/controllers/catalog_controller.rb]: Hmm. I don't know what that's doing there... probably should be removed...

Comment (cam156) [inline app/controllers/catalog_controller.rb]: @awead so is `title_name = 'title_ssi'` still needed?  I thought it could be deleted

**Comment by cam156** [comment]
@awead looks like your build got caught by the cucumber test that is flaky.  I think the line I mentioned above is no longer needed.  Since you are changing the code cleaning up lines makes sense to me.  Once that line is removed and the build passes I think this PR is good to go.

**Comment by awead** [comment]
@cam156 the pf key is used in Scholarsphere.  I could try omitting it but I just used the same config.

**Comment by cam156** [comment]
@awead what I am saying is that is a garbage do nothing line.  Why keep it?

**Comment by awead** [comment]
@cam156 amended.

**Comment by awead** [comment]
@cam156 pf boosts the relevance of returned documents if the terms are found in close proximity:
https://wiki.apache.org/solr/DisMaxQParserPlugin#pf_.28Phrase_Fields.29
So, if the search terms are present in the title, the relevance of the search result will be boosted higher than other search results that have the terms in different fields. We're still using the same technique in Scholarsphere:
https://github.com/psu-stewardship/scholarsphere/blob/develop/app/controllers/catalog_controller.rb#L152

**Comment by cam156** [comment]
Once the build passes I will :shipit:

**Pull request merged**